controller/src/enhancements/spectators_list.rs

```diff
@@ -0,0 +1,166 @@
+use std::{
+    ffi::CStr,
+};
+
+use anyhow::Context;
+use cs2::CEntityIdentityEx;
+use obfstr::obfstr;
+use cs2_schema_generated::cs2::client::{
+    CCSPlayerController,
+    C_CSObserverPawn,
+};
+
+use super::Enhancement;
+
+pub struct SpectatorInfo {
+    pub spectator_name: String,
+}
+
+pub struct SpectatorsList {
+    spectators: Vec<SpectatorInfo>
+}
+
+impl SpectatorsList {
+    pub fn new() -> Self {
+        SpectatorsList{
+            spectators: Default::default(),
+        }
+    }
+}
+
+impl Enhancement for SpectatorsList {
+    fn update(&mut self, ctx: &crate::UpdateContext) -> anyhow::Result<()> {
+        self.spectators.clear();
+
+        if !ctx.settings.spectators_list {
+            return Ok(());
+        }
+
+        let local_player_controller = ctx
+            .cs2_entities
+            .get_local_player_controller()?
+            .try_reference_schema()
+            .with_context(|| obfstr!("failed to read local player controller").to_string())?;
+
+        let local_player_controller = match local_player_controller {
+            Some(controller) => controller,
+            None => {
+                /* We're currently not connected */
+                return Ok(());
+            }
+        };
+
+        let observice_entity_handle = if local_player_controller.m_bPawnIsAlive()? {
+            local_player_controller.m_hPawn()?.get_entity_index()
+        } else {
+            return Ok(());
+        };
+
+        for entity_identity in ctx.cs2_entities.all_identities() {
+            if entity_identity.handle::<()>()?.get_entity_index() == observice_entity_handle {
+                /* current pawn we control/observe */
+                continue;
+            }
+
+            let entity_class = ctx
+                .class_name_cache
+                .lookup(&entity_identity.entity_class_info()?)?;
+
+            if !entity_class
```

Comment:
Why are you filtering for `CCSPlayerController` when you afterwards get the pawn again?  
You can just loop over all pawns (`C_CSPlayerPawn`). Doing this, the first check "current pawn we control/observe" would also be affective.

Reply:
Updated.